### PR TITLE
Clean up '#if CORE' in FileSystemProvider.cs

### PR DESCRIPTION
**Pull request by iSazonov**
Related #3565 and #4357.

### CODE REVIEW

File: src/System.Management.Automation/namespaces/FileSystemProvider.cs [hunk 1]
```diff
@@ -7966,26 +7966,6 @@ private struct REPARSE_GUID_DATA_BUFFER
             IntPtr OutBuffer, int nOutBufferSize,
             out int pBytesReturned, IntPtr lpOverlapped);
 
-#if !CORECLR
-
-        [DllImport("kernel32.dll", SetLastError = true, CharSet = CharSet.Unicode)]
-        private static extern IntPtr FindFirstFileName(
```
Comment (daxian-dbw) [inline src/System.Management.Automation/namespaces/FileSystemProvider.cs]: Again, need to understand what those APIs are for. Maybe they are for features that need to be brought back to powershell core.

Comment (iSazonov) [inline src/System.Management.Automation/namespaces/FileSystemProvider.cs]: Currently it is not used. The API is specific for Windows file systems. It works with hard links - it is already in PowerShell Core.

Comment (daxian-dbw) [inline src/System.Management.Automation/namespaces/FileSystemProvider.cs]: Then `private static List<string> InternalGetTarget(string filePath)`  needs to be revisited as it basically only handles Unix platforms. Maybe just remove it, and change `public static IEnumerable<string> GetTarget(PSObject instance)` to directly handle `Windows/Unix` with `if UNIX/else`.

**Comment by iSazonov** [comment]
I close th PR - we need in-depth review the API - perhaps it is not used but it is public.

**Pull request closed without merging**